Sources/SourceKitLSP/Swift/SwiftLanguageServer.swift

```diff
@@ -370,7 +381,10 @@ extension SwiftLanguageServer {
         range: .bool(true),
         full: .bool(true)),
       inlayHintProvider: .value(InlayHintOptions(
-        resolveProvider: false))
+        resolveProvider: false)),
+      diagnosticProvider: DiagnosticOptions(
+        interFileDependencies: true,
```

Comment:
How do you model the `interFileDependencies`? I don’t think sourcekitd keeps track of that in any way, so I would think that we should return `false` here.

Reply:
```
/**
	 * Whether the language has inter file dependencies meaning that
	 * editing code in one file can result in a different diagnostic
	 * set in another file. Inter file dependencies are common for
	 * most programming languages and typically uncommon for linters.
	 */
```

My read of this is that by setting `interFileDependencies: true`, we're telling the LSP client that it's worth re-querying diagnostics for file B after changes in file A since it might have fixed them, so this is what allows the behavior in the GIF in this PR summary.